### DISCUSSION TITLE
Issue when the response is a zip file, the reproduced zip file is invalid

### DISCUSSION THREAD

**Discussion by codingsett** [fastapi/fastapi#7279] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Description

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```@router.get("/files", tags=["files"])
async def all_files():
    zip_io = BytesIO()
    with zipfile.ZipFile(zip_io, mode='w', compression=zipfile.ZIP_DEFLATED) as backup_zip:
        for f in listdir(files_folder):
            filename = f.split(".")[0]
            extension = f.split(".")[-1]
            try:
                backup_zip.write(f'{filename}.{extension}', f'{filename}.{extension}')
            except:
                pass

        response = StreamingResponse(iter([zip_io.getvalue()]), media_type="application/x-zip-compressed")
        response.headers["Content-Disposition"] = "attachment; filename=application.zip"
        # response.headers["Content-Type"] = 'application/force-download'
        return response
```

### Description

So i searched all over from google to github, and applied all solutions even using fileresponse, the zip file produced is not corrupted, i double checked that while i was debugging(its produced on the computer by that code and i can extract it). The issue seems like its from the Response, FileResponse, StreamingResponse methods. Its like they change the final file content or invalidate it somehow. I have spent at least a full day on this and i came to the conclusion that the issue most likely is from fast api. Any help guys?
### Environment

* OS: [e.g. Linux / Windows]:
* FastAPI Version [0.61.1]:

* Python version: 3.7

### Additional context

This is what i get after i have downloaded the file:

![image](https://user-images.githubusercontent.com/27306293/92153923-a6f16b80-ee2d-11ea-89d3-d87738c8ccbf.png)

**Comment by henhuy** [comment]
For me, this here works (I'm writing CSV on-the-fl and zip it altogether - not the same, but maybe it helps):
```
import csv
import zipfile
from io import BytesIO, StringIO
from fastapi.responses import StreamingResponse

zipped_file = BytesIO()
with zipfile.ZipFile(zipped_file, 'a', zipfile.ZIP_DEFLATED) as zipped:
    csv_data = StringIO()
    writer = csv.writer(csv_data, delimiter=',')
    writer.writerow(["test", "data"])
    csv_data.seek(0)
    csv_buffer = csv_data.read()
    zipped.writestr(f"test_data.csv", csv_buffer)
zipped_file.seek(0)
response = StreamingResponse(zipped_file, media_type="application/x-zip-compressed")
response.headers["Content-Disposition"] = "attachment; filename=test.zip"
return response
```

**Comment by codingsett** [comment, marked as answer]
I found the issue was actually on the js side not fastapi. Thanks for the assist though.

**Comment by rudy-haryanto** [comment]
> I found the issue was actually on the js side not fastapi. Thanks for the assist though.

Can you please elaborate? I have the same issue on fastapi zip response

**Comment by TheBhat** [comment]
Hey.. I got the same issue when running FastAPI from APIGW+lambda. any solution for this issue?

**Comment by codingsett** [comment]
```
axios
        .get("/api/files/my-file", { responseType: 'arraybuffer' })
        .then(response => {
          const fileURL = window.URL.createObjectURL(new Blob([response.data]))
          const fileLink = document.createElement('ba')

          fileLink.href = fileURL
          fileLink.setAttribute('download', 'file.zip')
          document.body.appendChild(fileLink)

          fileLink.click()
        })
```
        
 Your frontend code should look something like this
